### ISSUE TITLE
component CRD v1beta1: Add capability for arbitrary security fields

### ISSUE THREAD

**Issue by csotiriou** [tmforum-oda/oda-ca#100]
I have noticed that in the v1beta1 specification of the `component` crd there are some properties missing compared to earlier versions, that would allows us to configure the security aspects of the component with greater ease.

Since it seems that the `security` contents inside the components are not yet final in terms of specification, I would like to propose / request the addition of the `x-kubernetes-preserve-unknown-fields` below the `security` field, so that custom kubernetes operators can be created until the `security` field is standardized.

The relevant field in the oda component CRD yaml is `spec.versions[4].schema.openAPIV3Schema.properties.spec.properties.security`

**Comment by LesterThomas** [comment]
> Since it seems that the `security` contents inside the components are not yet final in terms of specification, I would like to propose / request the addition of the `x-kubernetes-preserve-unknown-fields` below the `security` field, so that custom kubernetes operators can be created until the `security` field is standardized.

Good point - I will add the `x-kubernetes-preserve-unknown-fields` back in below the `security` field
